log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/rolling/RollingRandomAccessFileManagerTest.java

```diff
@@ -333,4 +334,40 @@ void testRolloverRetainsFileAttributes() throws Exception {
                 .permissions();
         assertEquals(filePermissions, actualFilePermissions);
     }
+
+    @Test
+    void testWriteHeaderWhenFileDoesNotExistBefore() throws IOException {
```

Comment:
I see that header write is not tested at all. Would you mind extending this to adapt the following footprint, please?

    @ParameterizedTest
    @CsvSource({
            "true,true",
            "true,false",
            "false,true",
            "false,false",
    })
    void testWriteHeaderWhenFileDoesNotExistBefore(final boolean append, final boolean fileExists) throws Exception {
        // ...
    }

If you think there are more cases (of header write) to be tested, please share.

Reply:
fix